tensorflow/compiler/mlir/tosa/transforms/fuse_bias_tf.cc

```diff
@@ -105,6 +101,35 @@ LogicalResult ConvertTFBiasAddOp::matchAndRewrite(
   rewriter.replaceOp(op, {result.getValue()});
 
   return success();
+
+  }
+
+  TF::Conv3DOp tf_conv3d_op =
+      dyn_cast_or_null<TF::Conv3DOp>(value.getDefiningOp());
+
+  if (tf_conv3d_op) {
+    // Sanity check to confirm rhs() has the expected shape of bias
+    auto filter_shape =
+      tf_conv3d_op.filter().getType().dyn_cast<RankedTensorType>().getShape();
+
+    // Bias dimension must match filter output channels, where tf.conv3d's filter
+    // is [D, H, W, I, O]
+    if (filter_shape.back() != bias_shape.back()) return failure();
+
+    llvm::Optional<Value> result = convertTFConv3DCommon(
+        rewriter, op, output_type, tf_conv3d_op.input(), tf_conv3d_op.filter(),
+        bias, tf_conv3d_op.strides(), tf_conv3d_op.dilations(),
+        tf_conv3d_op.padding(), tf_conv3d_op.data_format());
+
+    if (!result) return failure();
+
+    rewriter.replaceOp(op, {result.getValue()});
```

Comment:
value() - getValue has been marked as deprecated.

Reply:
Good to know.  Changed.